splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java

```diff
@@ -59,6 +80,86 @@ public void showOnLaunch(final AppCompatActivity activity) {
+            () -> {
+                androidx.core.splashscreen.SplashScreen windowSplashScreen = androidx.core.splashscreen.SplashScreen.installSplashScreen(
+                    activity
+                );
+                windowSplashScreen.setKeepOnScreenCondition(() -> isVisible || isHiding);
+
+                // Set Fade Out Animation
+                windowSplashScreen.setOnExitAnimationListener(
+                    windowSplashScreenView -> {
+                        final ObjectAnimator fadeAnimator = ObjectAnimator.ofFloat(windowSplashScreenView.getView(), View.ALPHA, 1f, 0f);
+                        fadeAnimator.setInterpolator(new LinearInterpolator());
+                        fadeAnimator.setDuration(settings.getFadeOutDuration());
+
+                        fadeAnimator.addListener(
+                            new AnimatorListenerAdapter() {
+                                @Override
+                                public void onAnimationEnd(Animator animation) {
+                                    isHiding = false;
+                                    windowSplashScreenView.remove();
+                                }
+                            }
+                        );
+
+                        fadeAnimator.start();
+
+                        isHiding = true;
+                        isVisible = false;
+                    }
+                );
+
+                // Set Pre Draw Listener & Delay Drawing Until Duration Elapses
+                final View content = activity.findViewById(android.R.id.content);
+
+                content
+                    .getViewTreeObserver()
+                    .addOnPreDrawListener(
+                        new ViewTreeObserver.OnPreDrawListener() {
+                            @Override
+                            public boolean onPreDraw() {
+                                // Run fade out animation since content is ready to draw
+                                if (settings.isAutoHide()) {
+                                    isVisible = true;
+                                    return true;
+                                }
+
+                                // Start Timer On First Run
+                                if (!isVisible && !isHiding) {
+                                    isVisible = true;
+
+                                    new Handler(context.getMainLooper())
+                                        .postDelayed(
+                                            () -> {
+                                                // Splash screen is done... start drawing content.
+                                                isVisible = false;
+                                                content.getViewTreeObserver().removeOnPreDrawListener(this);
+                                            },
+                                            settings.getShowDuration()
+                                        );
+                                }
```

Comment:
```suggestion
                                if (settings.isAutoHide()) {
                                    // Start Timer On First Run
                                    if (!isVisible && !isHiding) {
                                        isVisible = true;
                                        new Handler(context.getMainLooper())
                                            .postDelayed(
                                                () -> {
                                                    // Splash screen is done... start drawing content.
                                                    isVisible = false;
                                                     content.getViewTreeObserver().removeOnPreDrawListener(this);
                                                },
                                                settings.getShowDuration()
                                            );
                                        }
                                    }                                
```
isAutoHide means the splash will automatically hide after the getShowDuration time, with current code it hides right away without waiting, I think the code that does the delayed hide should be inside the isAutoHide() check.